### DISCUSSION TITLE
MIDI channel assignment not working properly

### DISCUSSION THREAD

**Discussion by PaulKob** [GrandOrgue/grandorgue#1829] [Help]
I'm a total newbie to GrandOrgue with a frustrating problem, having trouble getting MIDI channel changes to work properly.

Setup: One keyboard, MacbookPro+MacOS 12.7, USB cable and GrandOrgue v3.13.2-1 (2021)

I have one digital keyboard where I created three registrations M1, M2 and M3, one each for channels 1, 2 and 3 respectively. I then added them to favorite buttons on the keyboard for easy quick access to organs with multiple manuals. I use a USB A-B cable between the keyboard and the MacbookPro.

I tested the connection with a MIDI monitor program and verified that the registrations transmit the proper messages to the desired channels. The MIDI program reports:
Channel - Data
16 - Bank Select 85
16 - Bank Select (fine) 1
16 - 9
01  MIDI note on
01  MIDI note off
etc.
This is the same for all three channels except  channel 2 shows 16 - 8, and channel 3 shows 16 - 7.

Loading the demo organ and setting the MIDI parameters in GO, with key range detected and setting the first manual to receive on channel 1 and the second on 2, (and saving the settings) I select favorite button 1, (M1-channel 1) on the keyboard. But when I play the keyboard the notes play on manual 2  with receiving channel still set to 2! I  cannot get any notes to play on manual 1. No matter how I define the parameters this happens.

I've read all the MIDI discussions and can find no similar problem. The closest was #668, having problems with MIDI messages getting to GO when another MIDI program is open. The response was to make sure a MIDI monitor program was quit before running GO, which I did.  But this was specifically for Windows 10.

Any help or suggestions would be helpful.
Thanks, Paul

**Comment by ClarkeBrunt** [comment]
You can log midi messages in GO - it's an option in one of the menus. What is being received? And show us the midi configuration for each manual - right-click them on the screen. That other thread mentioned 'Initial midi configuration' - don't use that for the moment - it's just for a best-guess default for a newly loaded organ.

**Comment by PaulKob** [comment]
Clarke,
The "Midi-Settings for Manual" window shows the settings I expect, channel 1 for first manual and 2 for second. The laptop is receiving the correct program control messages for the channels from the keyboard. However the GO MIDI log shows the receiving channels are 2 and 3! That explains why manual 1 gets nothing and channel 1 out goes to channel 2 in GO. But why?

**Comment by ClarkeBrunt** [comment]
Midi channels are normally called 1-16, but the number encoded in the binary message is actually 0-15 (4 bits). GO is using the 1-16 convention; it sounds as though you are thinking of 0-15, so off by one.

So change to 2 and 3 in the dialog. But you don't normally fill these in by typing them - you right-click on the manual, then "Detect complex...". (this is preferred to the simple detect) and follow the displayed instructions. The it will be set to what is actually received, rather than what you think will be received.

**Comment by PaulKob** [comment]
Clarke,
That worked. 
Thanks for the help
Paul
